clang/lib/Driver/Driver.cpp

```diff
@@ -821,7 +828,13 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
           std::pair<StringRef, StringRef> I = Val.split(':');
           if (!I.first.empty() && !I.second.empty()) {
             llvm::Triple TT(I.first);
-            if (TT.getArch() == llvm::Triple::UnknownArch || !TT.isSPIR()) {
+            // Check for invalid triple values.
+            StringRef A(TT.getArchName());
+            if (TT.getArch() == llvm::Triple::UnknownArch || !TT.isSPIR() ||
+                (TT.getSubArch() == llvm::Triple::NoSubArch &&
+                 ((TT.getArch() == llvm::Triple::spir && !A.equals("spir")) ||
+                  (TT.getArch() == llvm::Triple::spir64 &&
+                   !A.equals("spir64"))))) {
```

Comment:
A usual "could-this-evaluation-be-factored-out". I see a couple options:
1) Well, factoring out the check into a separate helper
2) Adding the check into `MakeSYCLDeviceTriple` and reusing it for the full triples creation here and at the `OffloadingActionBuilder` level. This is an arguable alternative - extends the functional changes by allowing incomplete triples, and maybe even results in double-message errors.
And overlaps with my current work :)